base/compiler/typelimits.jl

```diff
@@ -433,6 +433,8 @@ function tmerge(lattice::ConditionalsLattice, @nospecialize(typea), @nospecializ
         end
         return Bool
     end
+    typea = widenconditional(typea)
+    typeb = widenconditional(typeb)
```

Comment:
They were already handled above (otherwise widening would be illegal for the lattice operation here)

Reply:
Ah, I realize we don't actually handle them above. Shouldn't we need to?

Reply:
Not if e.g. `isa(typea, Conditional)`, `typeb === Bool`

Reply:
> Ah, I realize we don't actually handle them above. Shouldn't we need to?

I don't think so - it's always legal to widen here. Did you have something specific in mind?

Reply:
Okay. No, I was just assuming it needed to be explicitly handled for the stacking of the lattices to work right.

Reply:
Most of the cases are handled by the `tmerge_fast_path` at the top of the lattice. It's currently not technically a fast path, because `tmerge` will not work properly without it. We should probably refactor things such that there is one entry point that always does that and then defers to what is currently tmerge for the actual merging, but where we can assume that the subtype cases have all been taken care of.